servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,482 @@
+    //         t6        |-- cut off time --------  |-- cut off time --------    |
+    //         t7        |  TABLE_ONE : 44          |                            |
+    //         t8        |                          |  TABLE_TWO : 44            |
+    //         t9        |                          | assign main to this branch |
+    //         t10       |  delete branch           |                            |
+
+    String prefix = "multiRefAssignAndDropRef";
+    Map<String, Map<String, Set<Content>>> expectedExpired = new HashMap<>();
+
+    // 'before' is same as main branch
+    Branch before = createBranch(prefix + "_0");
+    Branch branch1 = createBranch(prefix + "_1");
+    Branch branch2 = createBranch(prefix + "_2");
+    // commit for TABLE_ONE on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, CID_ONE, TABLE_ONE, METADATA_ONE, null, null);
+    // expect this commit to be expired as it is before cutoff time and not the head commit
+    fillExpectedContents(Branch.of(branch1.getName(), table1.hash), 1, expectedExpired);
+    // commit for TABLE_TWO on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, CID_TWO, TABLE_TWO, METADATA_ONE, null, null);
+    // expect this commit to be expired as it is before cutoff time and not the head commit
+    fillExpectedContents(Branch.of(branch2.getName(), table2.hash), 1, expectedExpired);
+
+    table1 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            table1.hash,
+            43,
+            CID_ONE,
+            TABLE_ONE,
+            METADATA_TWO,
+            table1.content,
+            null);
+    table2 =
+        commitSingleOp(
+            prefix,
+            branch2,
+            table2.hash,
+            43,
+            CID_TWO,
+            TABLE_TWO,
+            METADATA_TWO,
+            table2.content,
+            null);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for TABLE_ONE on branch 1
+    table1 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            table1.hash,
+            44,
+            CID_ONE,
+            TABLE_ONE,
+            METADATA_THREE,
+            table1.content,
+            null);
+
+    // commits for TABLE_TWO on branch2
+    table2 =
+        commitSingleOp(
+            prefix,
+            branch2,
+            table2.hash,
+            44,
+            CID_TWO,
+            TABLE_TWO,
+            METADATA_THREE,
+            table2.content,
+            null);
+
+    // assign main to branch2 should not affect as it is performed after cutoff.
+    getApi()
+        .assignBranch()
+        .branch(Branch.of(branch2.getName(), table2.hash))
+        .assignTo(before)
+        .assign();
+
+    // drop ref branch1 should not affect as it is performed after cutoff timestamp.
```

Comment:
I suspect there should also be a config for the cutoff-time for content on dead references.

Reply:
the current configuration works for dead references also including the commit protection time config.

Reply:
Dead references might not be expected to be retained for the duration of the default cutoff-time, which can be quite long. So a separate config makes sense.

Reply:
we anyway support per reference configuration. We can configure cutoff time for dead references using that ?

Reply:
we can also have it in a follow up if it is really required.